homeassistant/components/asuswrt/bridge.py

```diff
@@ -47,6 +59,42 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+async def migrate_legacy_protocols(
```

Comment:
I'm not sure why this is here? If the flow version doesn't match, you trigger an re-auth. Why doesn't that one take care of this all?

Reply:
This is the main point. `re-auth` is triggered only when we have an `ssh` configuration that use `ssh` key instead of password, in other case migration can be completed automatically at startup because password do not changed. 
Also the way used to update version in config flow during `re-auth` probably is not really correct. 
@gjohansson-ST was suggesting to remove migration with `re-auth`, and so leave `ssh` configuration in that case. It make sense but I'm wondering if user should be informed about this.